pkg/utils/overrider/overrider_test.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package overrider
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1alpha1 "go.goms.io/fleet/apis/placement/v1alpha1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+func TestIsClusterMatched(t *testing.T) {
+	tests := []struct {
+		name    string
+		cluster clusterv1beta1.MemberCluster
+		rule    placementv1alpha1.OverrideRule
+		want    bool
+	}{
+		{
+			name: "matched overrides with empty cluster label",
+			cluster: clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster-1",
+				},
+			},
+			rule: placementv1alpha1.OverrideRule{}, // empty cluster label selects all clusters
+			want: true,
+		},
+		{
+			name: "rule with empty cluster selector",
+			cluster: clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster-1",
+				},
+			},
+			rule: placementv1alpha1.OverrideRule{
+				ClusterSelector: &placementv1beta1.ClusterSelector{},
```

Comment:
Now get your point after reading the next comment

// A label selector is a label query over a set of resources. The result of matchLabels and
// matchExpressions are ANDed. An empty label selector matches all objects. A null
// label selector matches no objects.
// +structType=atomic
type LabelSelector struct {

labelSelector is doing the reverse way. i can keep them consistent, otherwise, it may too confusing.